# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/PeerConnectionCtx.h
# Commit: 3cbde68f2f7d
# Full Hash: 3cbde68f2f7d3d6482696d160db00b62fa6a5b30
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2022-12-22 21:34:53
# Regressor Bug: 1800136
# File Overlap Count: 1
# Description:
#   Bug 1800136: Fix gap between PC.close() and the storage of its final stats in PeerConnectionCtx. r=ng
#   
#   Ensures that:
#   1) We never have stats report in a still-registered PC _and_ the stats for
#      closed PCs.
# ==============================================================================

diff -r fef019ffca5f -r 3cbde68f2f7d dom/media/webrtc/jsapi/PeerConnectionCtx.h
--- a/dom/media/webrtc/jsapi/PeerConnectionCtx.h	Thu Dec 22 15:29:02 2022 +0000
+++ b/dom/media/webrtc/jsapi/PeerConnectionCtx.h	Thu Dec 22 15:29:02 2022 +0000
@@ -121,11 +121,6 @@
 
   SharedWebrtcState* GetSharedWebrtcState() const;
 
-  // WebrtcGlobalInformation uses this; we put it here so we don't need to
-  // create another shutdown observer class.
-  mozilla::dom::Sequence<mozilla::dom::RTCStatsReportInternal>
-      mStatsForClosedPeerConnections;
-
   void RemovePeerConnection(const std::string& aKey);
   void AddPeerConnection(const std::string& aKey,
                          PeerConnectionImpl* aPeerConnection);
@@ -133,6 +128,8 @@
   template <typename Function>
   void ForEachPeerConnection(Function&& aFunction) const;
 
+  void ClearClosedStats();
+
  private:
   std::map<const std::string, PeerConnectionImpl*> mPeerConnections;
 